### DISCUSSION TITLE
Advice for GSoC 2024: Projects 10/11, and C++

### DISCUSSION THREAD

**Discussion by dante-hl** [openvinotoolkit/openvino#23615] [Google Summer of Code]
Hello there,

I am a Master's student studying Data Science at the University of Pennsylvania, with plans of pursuing further graduate study. My background and interests are in deep learning theory, but I've recently become quite interested in the engineering aspects of deep learning, given that many projects at the frontier of research are an equal blend of research and engineering. In particular, I'm hoping to gain better understanding of the building blocks of production-ready systems and after some research it seems OpenVINO is the perfect place to do that!

The projects that I'm interested in are:
Project 10: Accelerating PyTorch Lightning and ComfyUI with torch.compile OpenVINO
Project 11: PyTorch Model Optimizations with torch.compile OpenVINO Backend

I have around 2-3 years of experience working in PyTorch and Python, using them in a variety of areas, eg. building Bayesian Inference Samplers, building Transformer Encoders, etc., and am hoping to through these projects quickly ramp up a working knowledge of C++ , as from experience I learn best by doing. That said, I realize my background in C++ may not be enough to properly get this done, so I'm happy to hear any advice for if there are more suitable projects.

Thank you for your time!

**Comment by rkazants** [comment]
@cavusmustafa, @ynimmaga, @suryasidd, please advice

**Comment by cavusmustafa** [comment]
Hi @dante-hl, thank you for your interest. Could you please prepare and email me the application based on the [Application Template](https://github.com/openvinotoolkit/openvino/wiki/Google-Summer-Of-Code#application-template)? We can provide feedback and you could submit it to GSoC.

Please feel free to ask any questions.

Thank you,
Mustafa
